### PR TITLE
[BUGFIX] `argilla server`: prevent errors when returning users progress

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR fixes the error when getting dataset progress by users with partial record annotations (only pending or completed). In those cases, the server raises an error since `pending` or `responses` in the response schema cannot be filled. 

The solution is to normalize and always send the same response structure, filling values with `0`'s.

Refs:  This error was found when working on https://github.com/argilla-io/argilla/pull/5479, which also includes the fix

**Type of change**

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Refactor (change restructuring the codebase without changing functionality)
- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-server/CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,10 @@ These are the section headers that we use:
 - Added [`rq`](https://python-rq.org) library to process background jobs using [Redis](https://redis.io) as a dependency. ([#5432](https://github.com/argilla-io/argilla/pull/5432))
 - Added a new background job to update records status when a dataset distribution strategy is updated. ([#5432](https://github.com/argilla-io/argilla/pull/5432))
 
+### Fixed
+
+- Fixed error when computing dataset progress by users without pending or completed infor ([#5484](https://github.com/argilla-io/argilla/pull/5484))
```
Comment (jfcalvo) [inline argilla-server/CHANGELOG.md]: `infor...` I guess you didn't finish the sentence.

File: argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py [hunk 2]
```diff
@@ -49,18 +49,96 @@ async def test_get_dataset_users_progress(self, async_client: AsyncClient, owner
             "users": [
                 {
                     "username": user_with_submitted.username,
-                    "completed": {"submitted": 3},
-                    "pending": {"submitted": 2},
+                    "completed": {"submitted": 3, "draft": 0, "discarded": 0},
+                    "pending": {"submitted": 2, "draft": 0, "discarded": 0},
```
Comment (jfcalvo) [inline argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py]:
```suggestion
                    "completed": {"submitted": 3, "discarded": 0, "draft": 0},
                    "pending": {"submitted": 2, "discarded": 0, "draft": 0},
```

File: argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py [hunk 3]
```diff
@@ -49,18 +49,96 @@ async def test_get_dataset_users_progress(self, async_client: AsyncClient, owner
...
-                    "pending": {"draft": 2},
+                    "completed": {"draft": 3, "submitted": 0, "discarded": 0},
+                    "pending": {"draft": 2, "submitted": 0, "discarded": 0},
                 },
                 {
                     "username": user_with_discarded.username,
-                    "completed": {"discarded": 3},
-                    "pending": {"discarded": 2},
+                    "completed": {"discarded": 3, "submitted": 0, "draft": 0},
+                    "pending": {"discarded": 2, "submitted": 0, "draft": 0},
```
Comment (jfcalvo) [inline argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py]:
```suggestion
                    "completed": {"submitted": 0, "discarded": 3, "draft": 0},
                    "pending": {"submitted": 0, "discarded": 2, "draft": 0},
```

File: argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py [hunk 4]
```diff
@@ -49,18 +49,96 @@ async def test_get_dataset_users_progress(self, async_client: AsyncClient, owner
...
+
+        response = await async_client.get(self.url(dataset.id), headers=owner_auth_header)
+
+        assert response.status_code == 200, response.json()
+        assert response.json() == {
+            "users": [
+                {
+                    "username": user_with_submitted.username,
+                    "completed": {"submitted": 0, "draft": 0, "discarded": 0},
+                    "pending": {"submitted": 2, "draft": 0, "discarded": 0},
```
Comment (jfcalvo) [inline argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py]:
```suggestion
                    "completed": {"submitted": 0, "discarded": 0, "draft": 0},
                    "pending": {"submitted": 2, "discarded": 0, "draft": 0},
```

File: argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py [hunk 5]
```diff
@@ -49,18 +49,96 @@ async def test_get_dataset_users_progress(self, async_client: AsyncClient, owner
...
+                },
+                {
+                    "username": user_with_draft.username,
+                    "completed": {"draft": 0, "submitted": 0, "discarded": 0},
+                    "pending": {"submitted": 0, "draft": 2, "discarded": 0},
```
Comment (jfcalvo) [inline argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py]:
```suggestion
                    "completed": {"submitted": 0, "discarded": 0, "draft": 0},
                    "pending": {"submitted": 0, "discarded": 0, "draft": 2},
```

File: argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py [hunk 6]
```diff
@@ -49,18 +49,96 @@ async def test_get_dataset_users_progress(self, async_client: AsyncClient, owner
...
+                },
+                {
+                    "username": user_with_discarded.username,
+                    "completed": {"discarded": 0, "submitted": 0, "draft": 0},
+                    "pending": {"submitted": 0, "draft": 0, "discarded": 2},
```
Comment (jfcalvo) [inline argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py]:
```suggestion
                    "completed": {"submitted": 0, "discarded": 0, "draft": 0},
                    "pending": {"submitted": 0, "discarded": 2, "draft": 0},
```

File: argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py [hunk 7]
```diff
@@ -49,18 +49,96 @@ async def test_get_dataset_users_progress(self, async_client: AsyncClient, owner
...
+            "users": [
+                {
+                    "username": user_with_submitted.username,
+                    "completed": {"submitted": 3, "draft": 0, "discarded": 0},
+                    "pending": {"submitted": 0, "draft": 0, "discarded": 0},
+                },
+                {
+                    "username": user_with_draft.username,
+                    "completed": {"submitted": 0, "draft": 3, "discarded": 0},
+                    "pending": {"draft": 0, "submitted": 0, "discarded": 0},
```
Comment (jfcalvo) [inline argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py]:
```suggestion
                    "completed": {"submitted": 0, "discarded": 0, "draft": 3},
                    "pending": {"submitted": 0, "discarded": 0, "draft": 0},
```

File: argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py [hunk 8]
```diff
@@ -49,18 +49,96 @@ async def test_get_dataset_users_progress(self, async_client: AsyncClient, owner
...
+                },
+                {
+                    "username": user_with_discarded.username,
+                    "completed": {"submitted": 0, "draft": 0, "discarded": 3},
+                    "pending": {"discarded": 0, "submitted": 0, "draft": 0},
```
Comment (jfcalvo) [inline argilla-server/tests/unit/api/handlers/v1/datasets/test_get_dataset_users_progress.py]:
```suggestion
                    "completed": {"submitted": 0, "discarded": 3, "draft": 0},
                    "pending": {"submitted": 0, "discarded": 0, "draft": 0},
```

File: argilla-server/src/argilla_server/api/schemas/v1/datasets.py [hunk 9]
```diff
@@ -87,15 +87,15 @@ class DatasetProgress(BaseModel):
...
+    discarded: int = 0
+    draft: int = 0
 
 
 class UserProgress(BaseModel):
     username: str
-    completed: RecordResponseDistribution
-    pending: RecordResponseDistribution
+    completed: RecordResponseDistribution = RecordResponseDistribution()
+    pending: RecordResponseDistribution = RecordResponseDistribution()
```
Comment (jfcalvo) [inline argilla-server/src/argilla_server/api/schemas/v1/datasets.py:98]: Is this change really needed? You already added default values to `RecordResponseDistribution`.

**Review by jfcalvo — CHANGES_REQUESTED** [review]

File: argilla-server/src/argilla_server/api/schemas/v1/datasets.py (hunk 9 above)
Comment (frascuchon) [reply argilla-server/src/argilla_server/api/schemas/v1/datasets.py:98]: Yes. We need to create a default value for pending or completed. Otherwise, those attributes are always required.

**Review by jfcalvo — APPROVED** [review]
